src/Oracle/Oracle.cs

```diff
@@ -0,0 +1,377 @@
+            timer.Interval = 1000 * 60 * 3;
+            timer.Start();
+            timer.Elapsed += new ElapsedEventHandler(OnTimer);
+        }
+
+        protected override void Configure()
+        {
+            var config = GetConfiguration();
+            Wallet = new NEP6Wallet(Combine(PluginsDirectory, nameof(Oracle), config.GetSection("Wallet").Value));
+            Nodes = config.GetSection("Nodes").GetChildren().Select(p => p.Get<string>()).ToArray();
+            MaxTaskTimeout = TimeSpan.FromMilliseconds(double.Parse(config.GetSection("MaxTaskTimeout").Value));
+            Https.Timeout = int.Parse(config.GetSection("HttpTimeout").Value);
+            Https.AllowPrivateHost = bool.Parse(config.GetSection("AllowPrivateHost").Value);
+        }
+
+        [RpcMethod]
+        public JObject SubmitOracleResponse(JArray _params)
+        {
+            ECPoint oraclePub = ECPoint.Parse(_params[0].AsString(), ECCurve.Secp256r1);
+            ulong requestId = (ulong)_params[1].AsNumber();
+            byte[] txSign = _params[2].AsString().HexToBytes();
+            byte[] msgSign = _params[3].AsString().HexToBytes();
+
+            var data = oraclePub.ToArray().Concat(BitConverter.GetBytes(requestId)).Concat(txSign).ToArray();
+            if (!Crypto.VerifySignature(data, msgSign, oraclePub)) throw new RpcException(-100, "Invalid sign");
+
+            using var snapshot = Blockchain.Singleton.GetSnapshot();
+            AddResponseTxSign(snapshot, requestId, txSign, oraclePub);
+            return new JObject();
+        }
+
+        [ConsoleCommand("start oracle", Category = "Oracle", Description = "Start oracle service")]
+        private void OnStart(string password)
+        {
+            Wallet.Unlock(password);
+
+            using var snapshot = Blockchain.Singleton.GetSnapshot();
+            if (!CheckOracleAvaiblable(snapshot, out ECPoint[] oracles)) throw new ArgumentException("The oracle service is unavailable");
+            if (!CheckOracleAccount(oracles)) throw new ArgumentException("There is no oracle account in wallet");
+
+            Interlocked.Exchange(ref CancelSource, new CancellationTokenSource())?.Cancel();
+            new Task(() =>
+            {
+                while (CancelSource?.IsCancellationRequested == false)
+                {
+                    using (var snapshot = Blockchain.Singleton.GetSnapshot())
+                    {
+                        IEnumerator<(ulong RequestId, OracleRequest Request)> enumerator = NativeContract.Oracle.GetRequests(snapshot).GetEnumerator();
+                        while (enumerator.MoveNext() && !CancelSource.IsCancellationRequested)
+                            if (!PendingQueue.TryGetValue(enumerator.Current.RequestId, out OracleTask task) || task.Tx is null)
+                                ProcessRequest(snapshot, enumerator.Current.RequestId, enumerator.Current.Request);
+                    }
+                    Thread.Sleep(500);
```

Comment:
Maybe it's possible to queue these new tasks on their way to this module? Polling with whole pending queue enumeration on every loop may be problematic if there are a lot of requests to process.